### ISSUE TITLE
Upgrading to projects to use 7.x cannot find the dependencies in the maven repos

### ISSUE THREAD

**Issue by apatrida** [Kodein-Framework/Kodein-DI#314]
Any upgrade to 7.x cannot find the 7.x dependencies for jvm generics version.

Does not work with `jcenter()` repo, does not work with `"https://dl.bintray.com/kodein-framework/Kodein-DI/"` repo.  Hand checking some of these repos does not show 7.x published other than for some artifacts.

Note that dependency resolution with some (or all) Gradle versions does not allow some packages to have a version in one repo but then missing the other version in the same, this causes it to not look at the other repository.  This could be an issue as these projects use an older Gradle and I cannot update at the moment to check with later versions.  But having partial releases out there for 7.x in some repos might be breaking this.

**Comment by romainbsl** [comment]
Hi, in our migration guide that can be found on the home of Kodein-DI we explain how to upgrade 
https://kodein.org/Kodein-DI/?7.0/migration-6to7

TL;DR: Kodein-DI jvm generics doesn't exists anymore. https://kodein.org/Kodein-DI/?7.0/migration-6to7#_new_modules_2

dependencies are still available from `jcenter()`, however choosing the right way to declare them will depends on which version of Gradle you are using.

**- Gradle 6:**
`implementation("org.kodein.di:kodein-di:7.0.0")`

**- Gradle 5:**

```kotlin
// build.gradle.kts
implementation("org.kodein.di:kodein-di:7.0.0")
```

```kotlin
// settings.gradle.kts
enableFeaturePreview("GRADLE_METADATA")
```

**- Gradle 4:**
`implementation("org.kodein.di:kodein-di-jvm:7.0.0")`
